javatests/arcs/core/data/expression/ExpressionTest.kt

```diff
@@ -154,6 +176,113 @@ class ExpressionTest {
         assertThat(evalExpression<Number, Number>(expr, currentScope, "arg" to 1)).isEqualTo(28)
     }
 
+    @Test
+    fun test_evalutor_paxel_from() {
+        val fromExpr = from<Number>("p") on "numbers"
+        assertThat(
+            evalExpression<Sequence<Number>, Sequence<Number>>(fromExpr, currentScope).toList()
```

Comment:
The signature of evalExpression is:
```
fun <T, R> evalExpression(
    expression: Expression<T>,
    currentScope: Expression.Scope = mapOf<String, Any>().asScope(),
    vararg params: Pair<String, Any>
): R
```

I am wondering, when would `T` be different than `R` ? If expression is of type `T`, would the result of evaluating it always be of type `T` as well? Or maybe some `T or Error` thing?

Reply:
Done